spark/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -135,15 +135,22 @@ private[delta] class ConflictChecker(
    * the transaction as if it had started while reading the `winningCommitVersion`.
    */
   def checkConflicts(): CurrentTransactionInfo = {
+    // Protocol and metadata checks.
```

Comment:
nit: I think it'd be nice if we left the rationale for the ordering behind so the next developer isn't afraid to reorder as necessary.  something like "we should enforce specifications in the protocol first"

Reply:
done.